src/Palladium/Exception/FailGenerateToken.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace Palladium\Exception;
+
+use Palladium\Component\Exception as Exception;
+
+class FailGenerateToken extends Exception
+{
+    protected $code = 3030;
+    protected $message = 'palladium.error.fail-provide-entropy-level';
```

Comment:
Unless I have misunderstood what can cause it, the message should be "missing-randomness-source". Or something like that.

Reply:
Based on docs, there are 3 types of Throwable might be generated by bin2hex: 

- If an appropriate source of randomness cannot be found, an Exception will be thrown.
- If invalid parameters are given, a TypeError will be thrown.
- If an invalid length of bytes is given, an Error will be thrown.

I've update this error message to the one you suggest, - but maybe I should even catch throwable instead of Exception and use general error message ?

Reply:
It's not `bin2hex()`, but `random_bytes()`. And you can't really get the 2nd and 3rd error. Both of those are errors in code (length being passes as string or negative number, which can't happen). Only the standard exception, if something goes wrong with server config.

Reply:
you are right.